pulumi/backuprunner/src/cleanup.ts

```diff
@@ -0,0 +1,56 @@
+import k8s, { KubernetesObject } from '@kubernetes/client-node'
+
+export const cleanup = async (k8sApi: k8s.KubernetesObjectApi, sts: string, namespace: string, pvcList: string, backupCount: number) => {
+  var pvcCount = pvcList.split(',').length;
+  var pvcsToKeepCount = pvcCount*backupCount;
+
+  const snapshots = await getExistingVolumeSnapshots(k8sApi, sts, namespace);
+  console.log(`Backup count to keep - ${backupCount}. ${sts} consists of ${pvcCount} PVC's, so will keep ${pvcsToKeepCount} latest snapshots`)
+
+  if(snapshots.length <= pvcsToKeepCount){
+    console.log(`Not archiving old snapshots, too few results`)
+  }else {
+    await removeOldSnapshots(k8sApi, snapshots, pvcsToKeepCount)
+  }
+
+  console.log("Snapshot cleanup completed");
+}
+
+const getExistingVolumeSnapshots = async (k8sApi: k8s.KubernetesObjectApi, sts: string, namespace: string): Promise<KubernetesObject[]> => {
+  const snapshots = await k8sApi.list("snapshot.storage.k8s.io/v1", "VolumeSnapshot", namespace, undefined, undefined, undefined, undefined, `statefulset=${sts}`);
+  const items = snapshots.body.items
+  console.log(`Found ${items.length} snapshots for ${namespace}.${sts}`)
+  return items.sort(sortByCreationTimestampDesc)
+}
+
+const sortByCreationTimestampDesc = ((a: KubernetesObject, b: KubernetesObject) => (new Date(b.metadata!!.creationTimestamp!!).getTime()  - new Date(a.metadata!!.creationTimestamp!!).getTime()));
+
+const removeOldSnapshots = async (k8sApi: k8s.KubernetesObjectApi, snapshots: KubernetesObject[], pvcsToKeepCount: number) => {
+  await Promise.all(getSnapshotsToRemove(snapshots, pvcsToKeepCount).map(async vc => await deleteSnapshot(k8sApi, vc)))
+}
+
+const deleteSnapshot = async (k8sApi: k8s.KubernetesObjectApi, vc: KubernetesObject) => {
```

Comment:
what is `vc` supposed to stand for? might be better to just name `snapshot`?

Reply:
should've been a `vs`, `volumeSnapshot`. Will patch

Reply:
I would say generally speaking, our typescript variable naming is more verbose/explicit versus the golang shorthand. Even with the golang shorthand, it is just iterators or narrowly scoped variables. With the shorthand param and being passed through several functions/callbacks, it loses it's meaning a bit regardless imo

Reply:
...let's definitely avoid the go naming patterns :)